src/classes/PMT_PaymentCreator_TEST.cls

```diff
@@ -1033,4 +1033,141 @@ private with sharing class PMT_PaymentCreator_TEST {
+
+        UTIL_CurrencyCache.cachedMapOfRatesByCurrencyCode.put(mockCurrency.IsoCode, mockCurrency);
+
+        String closedWonStage = UTIL_UnitTestData_TEST.getClosedWonStage();
+        String openStage = UTIL_UnitTestData_TEST.getOpenStage();
+
+        UTIL_CustomSettingsFacade.getContactsSettings().Payments_Auto_Close_Stage_Name__c = closedWonStage;
+
+        //create test data
+        Account acct = new Account (
+            Name='Test Account'
+        );
+        insert acct;
+
+        // I have an opportunity for CAD100 in a stage of Prospecting.
+        Opportunity opp = new Opportunity (
+            Name='Non US Currency',
+            AccountId = acct.Id,
+            Amount = 100.00,
+            CloseDate = Date.today(),
+            StageName = openStage
+        );
+
+        opp.put('CurrencyIsoCode', mockCurrency.IsoCode);
+        insert opp;
+
+        npe01__OppPayment__c payment = Database.query(
+            'SELECT Id, CurrencyIsoCode, npe01__Paid__c, npe01__Payment_Amount__c, npe01__Payment_Date__c ' +
+            'FROM npe01__OppPayment__c ' +
+            'WHERE npe01__opportunity__c = \'' + opp.Id + '\'' +
+            'LIMIT 1'
+        );
+        payment.npe01__Paid__c = true;
+        payment.npe01__Payment_Date__c = Date.today();
+
+        update payment;
+
+        opp = [
+            SELECT StageName
+            FROM Opportunity
+            WHERE Id = :opp.Id
+            LIMIT 1
+        ];
+
+        System.assertEquals(closedWonStage,opp.StageName);
+    }
+
+    /*********************************************************************************************************
+    * @description
+        operation:
+            Create an Opportunity in an org where multicurrency is NOT enabled
+            with stage name Prospecting and an amount of 100 CAD
+            update the corresponding payment so it is Paid and have a date.
+        verify:
+            Opportunity stage changes from 'Prospecting' to 'Closed/Won'
+    **********************************************************************************************************/
+    @isTest
+    static void testOpportunityAutoClosingWithoutMulticurrency() {
+
+        if (UTIL_Currency.getInstance().isMultiCurrencyOrganization()) {
+            return;
+        }
+
+        UTIL_Currency_TEST.UtilCurrencyMock mock = new UTIL_Currency_TEST.UtilCurrencyMock();
+        mock.isMultiCurrencyOrganizationReturn = false;
+        UTIL_Currency.instance = mock;
+
+        String closedWonStage = UTIL_UnitTestData_TEST.getClosedWonStage();
+        String openStage = UTIL_UnitTestData_TEST.getOpenStage();
+
+        UTIL_CustomSettingsFacade.getContactsSettings().Payments_Auto_Close_Stage_Name__c = closedWonStage;
+
+        //create test data
+        Account acct = new Account (
+            Name='Test Account'
+        );
+        insert acct;
+
+        // I have an opportunity for CAD100 in a stage of Prospecting.
```

Comment:
remove reference to CAD - we can't guarantee which currency we'll be in here

Reply:
Removed CAD from comment.